core/src/main/kotlin/fr/sncf/osrd/sim/DynReservationRoute.kt

```diff
@@ -0,0 +1,17 @@
+package fr.sncf.osrd.sim
+
+import fr.sncf.osrd.new_infra_state.api.ReservationRouteState
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+class DynReservationRoute {
+    private val _state = MutableStateFlow(ReservationRouteState.Summary.FREE);
+    val state = _state.asStateFlow()
+
+    fun forceStateChange(newState: ReservationRouteState.Summary) {
+        while (true) {
+            if (_state.compareAndSet(_state.value, newState))
+                break
+        }
```

Comment:
Do we really need the `compareAndSet` rather than just `update` here? If we do, I'm not sure if this loop is a solution